sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzurePath.java

```diff
@@ -774,10 +775,10 @@ private String rootToFileStore(String root) {
         return root.substring(0, root.length() - 1); // Remove the ROOT_DIR_SUFFIX
     }
 
-    static void ensureFileSystemOpen(Path p) throws IOException {
+    static void ensureFileSystemOpen(Path p) {
         if (!p.getFileSystem().isOpen()) {
             throw LoggingUtility.logError(((AzurePath) p).logger,
-                new IOException("FileSystem for path has been closed. Path: " + p.toString()));
+                new ClosedFileSystemException());
```

Comment:
Could we include a message for this exception to help troubleshooting

Reply:
This exception type actually doesn't allow you to set a message